src/Umbraco.Core/Services/ImportExport/TemporaryFileToXmlImportService.cs

```diff
@@ -0,0 +1,91 @@
+using System.Xml;
+using System.Xml.Linq;
+using Umbraco.Cms.Core.Models.ContentEditing;
+using Umbraco.Cms.Core.Models.ImportExport;
+using Umbraco.Cms.Core.Models.TemporaryFile;
+using Umbraco.Cms.Core.Services.OperationStatus;
+
+namespace Umbraco.Cms.Core.Services.ImportExport;
+
+public class TemporaryFileToXmlImportService : ITemporaryFileToXmlImportService
+{
+    private readonly ITemporaryFileService _temporaryFileService;
+    private readonly IPackageDataInstallation _packageDataInstallation;
+
+    public TemporaryFileToXmlImportService(
+        ITemporaryFileService temporaryFileService,
+        IPackageDataInstallation packageDataInstallation)
+    {
+        _temporaryFileService = temporaryFileService;
+        _packageDataInstallation = packageDataInstallation;
+    }
+
+    public async Task<Attempt<XElement?, TemporaryFileXmlImportOperationStatus>> LoadXElementFromTemporaryFileAsync(
+        Guid temporaryFileId, bool cleanupFile = true)
+    {
+        TemporaryFileModel? documentTypeFile = await _temporaryFileService.GetAsync(temporaryFileId);
+        if (documentTypeFile is null)
+        {
+            return Attempt.FailWithStatus<XElement?, TemporaryFileXmlImportOperationStatus>(
+                TemporaryFileXmlImportOperationStatus.TemporaryFileNotFound, null);
+        }
+
+        var xmlDocument = new XmlDocument { XmlResolver = null };
+        await using (Stream fileStream = documentTypeFile.OpenReadStream())
+        {
+            xmlDocument.Load(fileStream);
+        }
+
+        var element = XElement.Parse(xmlDocument.InnerXml);
+
+        if (cleanupFile)
+        {
+            await _temporaryFileService.DeleteAsync(documentTypeFile.Key);
+        }
+
+        return Attempt.SucceedWithStatus<XElement?, TemporaryFileXmlImportOperationStatus>(
+            TemporaryFileXmlImportOperationStatus.Success,
+            element);
```

Comment:
Is there a reason why we can't load the XML directly into an `XDocument` here?

```suggestion
        XDocument document;
        await using (Stream fileStream = documentTypeFile.OpenReadStream())
        {
            document = await XDocument.LoadAsync(fileStream, LoadOptions.None, CancellationToken.None);
        }

        if (cleanupFile)
        {
            await _temporaryFileService.DeleteAsync(documentTypeFile.Key);
        }

        return Attempt.SucceedWithStatus(TemporaryFileXmlImportOperationStatus.Success, document.Root);
```